Updated showcase steps with descriptions (#2)
identity-com/docs: showcase/README.md

```

### SOL DID Client
The SOL DID client allows you to manage DID documents on Solana.

```shell
$ yarn global add @identity.com/sol-did-client
```

### Cryptid Client

> The Cryptid client allows you to manage your cryptid account on the cli.

Reviewer: What is a cryptid account? Some short details or more links needed here IMO

```shell
$ yarn global add @identity.com/cryptid-cli

$ cryptid --help
```

### Gateway Client

> The gateway client allows managing of gateway tokens.

Reviewer: What are gateway tokens?

```

### SOL DID Client
The SOL DID client allows you to manage DID documents on Solana.

```shell
$ yarn global add @identity.com/sol-did-client
```

### Cryptid Client
The Cryptid client allows you to manage your Cryptid account on the cli.

[See here](https://github.com/identity-com/cryptid) for more information on Cryptid.

```shell
$ yarn global add @identity.com/cryptid-cli

$ cryptid --help
```

### Gateway Client

> The gateway client allows managing of gateway tokens. A gateway token is required for a user to prove that they have

Reviewer: ```suggestion
The gateway client allows the management of "Gateway Tokens". A gateway token is required for a user to prove that they have
```

been verified by a gatekeeper before being allowed to use certain on-chain applications. The rules required for a 
gatekeeper to issue a token could include things like age checks, OFAC checks, etc.

> More information can be found [here](https://github.com/identity-com/on-chain-identity-gateway).

Reviewer: this might be a better link https://identity-com.github.io/on-chain-identity-gateway

$ cryptid key add <pubkey> key2

# Transfer SOL to the new key
$ cryptid transfer <pubkey> 1000000

# View the updated DID document (with the additional key added)
$ cryptid document

# Update cryptid configuration to use the new key by default
$ cryptid config set keyFile $(pwd)/key2.json

# Remove the default key
$ cryptid key remove default

# Check the updated DID document
$ cryptid document

# Update cryptid configuration to use the original (default) key
$ cryptid config set keyFile $HOME/.config/solana/id.json

# Expect transfer of using the removed key to fail
$ cryptid transfer <pubkey> 1000000

# Switch back to your new key
$ cryptid config set keyFile $(pwd)/key2.json
```

## Controllers

Here we show how you can use [Cryptid](https://github.com/identity-com/cryptid) to setup one DID to be a controller of 
another DID. For example, if a DID represents a business in the real world, the CEO's DID can be setup as a 
controller of the business. This will allow the CEO to sign as that business using their own keys.

You can go through the steps below and follow along on YouTube: https://www.youtube.com/watch?v=4oSTRnvnmNM

```shell

> # Create a new key to demonstrate the controlled cryptid account

Reviewer: "demonstrate" here is not really telling us much - we are creating a new DID and for that we need to first generate a key that the did will be derived from

$ solana-keygen new -o controlled.json

> # Create a cryptid configuration for the new key

Reviewer: ```suggestion
# Create a cryptid configuration for the new DIDI
```

Reviewer: Also Cryptid should probably be capitalised across the board as it is a proper noun.

Reviewer: Also indicate why you are naming the new config controlled.yml. This new did will be the one we are going to control.

$ cryptid init -k controlled.json -p controlled.yml

> # View the configuration

Reviewer: check the configuration - with the -c controlled.yml you are instructing Cryptid to use the config of the new did instead of the config at $HOME/.config/cryptid/config.yml (please check this path)

Author: Is this not intentional? i.e. to show the configuration for the config of the new did ? (we show the default one earlier on)

Reviewer: yes sorry this is my suggestion of a more informative comment.  (Please paraphrase)

$ cryptid config -c controlled.yml

# Fund new cryptid address
$ cryptid transfer <NEW_DID> 100000000

> # Fund Signer #default signer

Reviewer: Probably need more details here - i.e why this needs to be funded - we are doing it just so we can add the controller DID

$ cryptid transfer <pubkey controlled.json> 100000000

> # Add your main account DID 

Reviewer: This is the important bit - we are adding the main account as a controller of the new did

$ cryptid controller add -c controlled.yml <OLD_DID>

# Add an alias for the controlled did (for convenience)
$ cryptid alias controlled <NEW_DID>

# Add an alias for your original did (for convenience)
$ cryptid alias me <OLD_DID>

# Check the DID document for the controlled cryptid

> $ cryptid document --as controlled

Reviewer: you are introducing the new --as parameter here - best to explain it.  --as differs from -c in that you are performing actions using your default did but controlling the other did. This becomes more powerful below.

# Check the balance for the controlled cryptid
$ cryptid balance --as controlled

> # Transfer SOL on behalf of the controlled cryptid account using the original cryptid account

Reviewer: "On behalf of" is a little confusing here because the funds are leaving the controlled account
"using the original cryptid account" is also a little hard to understand- what does "using " mean exactly?

## Getting started with did:sol

Here we show how to display your Solana address using the Solana client, and displaying your DID document
using the [sol-did](https://github.com/identity-com/sol-did) cli.

You can follow the steps below and follow along on YouTube: https://www.youtube.com/watch?v=G4vYa0uxYcs

```shell
# Display your Solana address
$ solana address

> # Display the DID document

Reviewer: A DID is automatically derived from any solana public key

$ cryptid document

# Update Cryptid configuration to use the original (default) key
$ cryptid config set keyFile $HOME/.config/solana/id.json

# Expect transfer of using the removed key to fail
$ cryptid transfer <pubkey> 1000000

# Switch back to your new key
$ cryptid config set keyFile $(pwd)/key2.json
```

## Controllers

Here we show how you can use [Cryptid](https://github.com/identity-com/cryptid) to setup one DID to be a controller of 
another DID. For example, if a DID represents a business in the real world, the CEO's DID can be setup as a 
controller of the business. This will allow the CEO to sign as that business using their own keys.

You can go through the steps below and follow along on YouTube: https://www.youtube.com/watch?v=4oSTRnvnmNM

```shell
# Create a key from which a new DID will be derived. We will be using this DID for the account we are going to control.
$ solana-keygen new -o controlled.json

# Create a Cryptid configuration for the new DID, called `controlled.yml` for the account we are going to control.
$ cryptid init -k controlled.json -p controlled.yml

# View the configuration of the controlled DID
$ cryptid config -c controlled.yml

> # Fund new Cryptid address so that we can add the controller DID

Reviewer: ```suggestion
# Fund the new DID's Cryptid account so that we can add the controller DID to it
```

$ cryptid transfer <NEW_DID> 100000000

# Fund the #default signer
$ cryptid transfer <pubkey controlled.json> 100000000

# Add your main account as a controller of the new DID 
$ cryptid controller add -c controlled.yml <OLD_DID>

# Add an alias for the controlled did (for convenience)
$ cryptid alias controlled <NEW_DID>

# Add an alias for your original did (for convenience)
$ cryptid alias me <OLD_DID>

# Check the DID document for the controlled Cryptid account. 
# By using `--as controlled`, we are instructing Cryptid to perform actions using your default DID while controlling
# the `controlled` DID.
$ cryptid document --as controlled

# Check the balance for the controlled Cryptid account
$ cryptid balance --as controlled

> # Perform a transfer transaction using your default account as the controlled account.

Reviewer: ```suggestion
# Perform a transfer transaction from the controlled account, signing with the default account.
```

## Gateway

> Here we will show how to issue a gateway token to a cryptid address from a dummy gatekeeper. This example will show

Reviewer: Probably need a line to explain what a dummy gatekeeper (or indeed a gatekeeper) is

how you cannot mont a NFT until a gateway token has been issued.

You can follow the steps below and follow along on YouTube: https://www.youtube.com/watch?v=4oSTRnvnmNM

Visit [http://candy.identity.com](http://candy.identity.com) and notice that you cannot yet mint an NFT.

```shell
# Issue a gateway token to your cryptid address
$ gateway issue -g $HOME/.config/solana/id.json -c devnet <CONTROLLED CRYPTID ADDRESS>
```

Visit [http://candy.identity.com](http://candy.identity.com) again, and notice that you can now mint an NFT.

## Credentials & Gateway

Here is an end-to-end showcase to show a "real life" example of a gateway token issuance based on credentials. This will

> show how you cannot mint and NFT until you've gone through a Civic KYC process, signed the credentials with cryptid

Reviewer: Best to mention (as we do in the video) the relationship between Civic and identity.com
